.github/workflows/build-and-release-maven.yml

```diff
@@ -91,12 +114,24 @@ jobs:
       - uses: actions/checkout@v3
         with:
           persist-credentials: false
-      - uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.36.0
+      - name: "Setup Java Build with specific configuration"
+        if: inputs.java-version && inputs.java-distribution
+        uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.36.0
+        with:
+          java-version: ${{ inputs.java-version}}
+          java-distribution: ${{ inputs.java-distribution}}
+      - name: "Setup Java Build with default configuration"
+        if: inputs.java-version == '' || inputs.java-distribution == ''
+        uses: Alfresco/alfresco-build-tools/.github/actions/setup-java-build@v1.36.0
       - uses: Alfresco/alfresco-build-tools/.github/actions/configure-git-author@v1.36.0
         with:
           username: ${{ secrets.BOT_GITHUB_USERNAME }}
           email: ${{ secrets.BOT_GITHUB_EMAIL }}
       - name: "Build"
         run: mvn -B -V install -DskipTests ${{ inputs.build-args }}
-      - name: "Release"
+      - name: "Release with specific version"
+        if: inputs.release-version && inputs.next-dev-version
+        run: mvn -B -DscmCommentPrefix="[maven-release-plugin][skip ci] " -Dusername="${{ secrets.BOT_GITHUB_USERNAME }}" -Dpassword="${{ secrets.BOT_GITHUB_TOKEN }}" -DskipTests -Darguments="-DskipTests ${{ inputs.release-args }}" -DreleaseVersion="${{ inputs.release-version }}" -DdevelopmentVersion="${{ inputs.next-dev-version }}" release:clean release:prepare release:perform
```

Comment:
you can avoid duplicating the entire command here, if you add a step before that will accumulate the additional `-Dprop` you need in a new variable, that will default to empty string and which you can use straight in the existing release step (let me know if you need additional pointers for achieving this)

Reply:
Sure. I will update it.

Reply:
Agreed with @gionn about avoid duplication, and also, I would go into a different direction and just offer additional maven props as an input, so that you can be more free to add custom additional parameters to the maven command line.

Said differently, instead of adding explicit inputs `release-version` and `next-dev-version` to the workflow, a generic `extra-maven-opts` option could make sense here, like it's done on https://github.com/Alfresco/alfresco-build-tools/blob/v1.36.0/.github/actions/maven-build-and-tag/action.yml#L144-L148 (with input defined at https://github.com/Alfresco/alfresco-build-tools/blob/v1.36.0/.github/actions/maven-build-and-tag/action.yml#L15-L17

Reply:
@atchertchian I completely agree with your points and will change it accordingly.